### PR TITLE
repo-updater: Explicity disallow external services from syncing on Cloud

### PR DESCRIPTION
**Pull request by ryanslade**
Instead of ignoring all site level external services from syncing on
Cloud we instead limit it to only exclude our "global" external services
as these are the ones we expect to have very large numbers of repos.

Might fix: https://github.com/sourcegraph/sourcegraph/issues/17424

### CODE REVIEW

File: cmd/repo-updater/shared/main.go [hunk 1]
```diff
@@ -162,10 +164,12 @@ func Main(enterpriseInit EnterpriseInit) {
 			case *schema.GitHubConnection:
 				if strings.HasPrefix(c.Url, "https://github.com") && c.Token != "" && c.CloudGlobal {
 					server.GithubDotComSource, err = repos.NewGithubSource(e, cf)
+					excludeFromSyncing = append(excludeFromSyncing, e.ID)
```
Comment (asdine) [inline cmd/repo-updater/shared/main.go:167]: Do you think we could do this filtering in the database instead? Like adding a column to the external services table and filtering them on l149

Comment (tsenart) [reply cmd/repo-updater/shared/main.go:167]: Yeah, that'd be cleaner.

Comment (ryanslade) [reply cmd/repo-updater/shared/main.go:167]: I thought about this when I originally added the "cloud global" idea. The problem is then that we need to manually update the database in order to make changes.

**Comment by ryanslade** [comment]
Hmm, I also just realised that we probably also want to index all synced repos.

File: cmd/repo-updater/shared/main.go (hunk 1 above)
Comment (ryanslade) [reply cmd/repo-updater/shared/main.go:167]: Also, it only applies to GitHub and GitLab

**Comment by ryanslade** [comment]
Closing in favour of https://github.com/sourcegraph/sourcegraph/issues/17472

**Pull request closed without merging**